homeassistant/components/netatmo/switch.py

```diff
@@ -0,0 +1,88 @@
+"""Support for Netatmo/BTicino/Legrande switches."""
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from pyatmo import modules as NaModules
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_URL_CONTROL, NETATMO_CREATE_SWITCH
+from .data_handler import HOME, SIGNAL_NAME, NetatmoDevice
+from .netatmo_entity_base import NetatmoBase
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Netatmo switch platform."""
+
+    @callback
+    def _create_entity(netatmo_device: NetatmoDevice) -> None:
+        entity = NetatmoSwitch(netatmo_device)
+        _LOGGER.debug("Adding switch %s", entity)
+        async_add_entities([entity])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, NETATMO_CREATE_SWITCH, _create_entity)
+    )
+
+
+class NetatmoSwitch(NetatmoBase, SwitchEntity):
+    """Representation of a Netatmo switch device."""
+
+    def __init__(
+        self,
+        netatmo_device: NetatmoDevice,
+    ) -> None:
+        """Initialize the Netatmo device."""
+        SwitchEntity.__init__(self)
+        super().__init__(netatmo_device.data_handler)
+
+        self._switch = cast(NaModules.Switch, netatmo_device.device)
+
+        self._id = self._switch.entity_id
+        self._attr_name = self._device_name = self._switch.name
```

Comment:
```suggestion
        self._attr_name = self._switch.name
```
I can't find any need for the self._device_name that can't be replaced by self._attr_name

Reply:
It is used in the base class though. E.g. https://github.com/home-assistant/core/blob/add_light/homeassistant/components/netatmo/netatmo_entity_base.py#L96